cmd/buildah/images.go

```diff
@@ -300,34 +306,63 @@ func outputImages(ctx context.Context, systemContext *types.SystemContext, store
 		}
 
 	outer:
-		for name, tags := range imageReposToMap(image.Names) {
-			for _, tag := range tags {
-				if !matchesReference(name+":"+tag, argName) {
+		for repo, tagsAndDigests := range imageReposToMap(image.Names) {
+			for _, tagOrDigest := range tagsAndDigests {
```

Comment:
AFAICS _nothing_ in here needs the data to actually be clustered by `repo`; it’s just two nested loops that process _all_ of the values of `image.Names` in sequence.

Is the goal only to sort the output by repo?

Don’t you really want a `[]reference.Named`, the full objects,, built by parsing `image.Names`, here? Then sort it using whatever key you need. And then process it in one non-nested loop.

Reply:
… sorry, the `imageTags` value in JSON output does only use the current-`repo` values. But should it do so?

Reply:
That's what I took your meaning to be in https://github.com/containers/buildah/pull/1972#discussion_r344352644.  Dropping the field, which would have been added in this PR.

Reply:
Once that field is dropped, I can’t see any more reason to have the `imageReposToMap` map and nesting two loops.

Reply:
Okay, reworked it.